### PR TITLE
Editing Toolkit Plugin: Prevent phpcbf errors from stopping sync process

### PR DESCRIPTION
**Pull request by jeyip**
#### Discussion

After running into issues syncing the editing toolkit plugin, we went to troubleshoot the problem in p1655945457500979-slack-CHN6J22MP and p1656098373502459-slack-C7YPUHBB2. 

In our discussions, however, we didn't understand why a phpcbf runtime error would halt the entire syncing process for ETK. Because our understanding is that `sync-newspack-blocks.sh` is a script meant for development purposes, in this PR, we remove logic that exits the syncing execution thread when a phpcbf error is thrown.

#### Proposed Changes

* _Do not_ exit when a phpcbf error is thrown while syncing newspack blocks

#### Testing Instructions

* Check out this PR
* In a terminal, navigate to `/apps/editing-toolkit` and run `yarn dev --sync`.
* Replace `apps/editing-toolkit/bin/sync-newspack-blocks.sh:135` with  `if true ; then` to force a phpcbf error 
* Verify that the syncing process does not end prematurely.

### CODE REVIEW

**Review by mattwiebe — APPROVED** [review]
Yes, please

File: apps/editing-toolkit/bin/sync-newspack-blocks.sh
Comment (creativecoder) [inline apps/editing-toolkit/bin/sync-newspack-blocks.sh:137]: What do you think about keeping the error and throwing it the build process (presumably used for production build), but not for the dev build, which should be a continuous process? (assuming there's an easy way to do this--I haven't looked yet.)

Comment (jeyip) [reply apps/editing-toolkit/bin/sync-newspack-blocks.sh:137]: > What do you think about keeping the error and throwing it the build process (presumably used for production build), but not for the dev build

This makes total sense to me, but I had assumed this was only used for the dev build ( might totally be mistaken ). If this is, in fact, used for prod, then I agree. Maybe @noahtallen or @fullofcaffeine can confirm?

Comment (fullofcaffeine) [reply apps/editing-toolkit/bin/sync-newspack-blocks.sh:137]: @noahtallen will know better, but I don't think this is (or should) be used for production releases. We use the `install-plugin.sh` from the sandbox, and the assets are fetched directly from the TC build associated with the PR. That being said, nothing prevents you from building a production release and syncing, so it might be useful for releasing to production in some edge cases?

Comment (fullofcaffeine) [reply apps/editing-toolkit/bin/sync-newspack-blocks.sh:137]: If we want to be extra-careful and still allow `phpcbf` to run when building for production, we can just test against `$NODE_ENV` like this:

```
if [ "$PHPCBF_ERRORED" = 1 ] && [ "$NODE_ENV" = "production" ] ; then                                                                                                                                           
   echo '!! There was an error executing phpcbf!'                                                                                                                                                                
   exit 1                                                                                                                                                                                                        
fi
```

Comment (creativecoder) [reply apps/editing-toolkit/bin/sync-newspack-blocks.sh:137]: There's a `release` mode referenced in the file, so I was inferring that this is used to make a production build of the blocks for Newspack sites (perhaps the file is poorly named?).

At least, I assume these blocks are still distributed separately for Newspack sites (in addition to being including in ETK), but maybe someone from that team can verify. @Automattic/newspack

Comment (fullofcaffeine) [reply apps/editing-toolkit/bin/sync-newspack-blocks.sh:137]: Ah good point, even though it's part of ETK, we're specifically talking about the newspack blocks syncing script, which seems to be part of a slightly different (and separate) release flow.

Comment (mattwiebe) [reply apps/editing-toolkit/bin/sync-newspack-blocks.sh:137]: Hmm thanks for digging in further, I naively thought this was just for sandbox sync and could be safely skipped.

We should probably just try to skip any/all functionality that isn't related to `yarn dev --sync` in this Newspack script and/or split out the concerns into separate scripts.

**Comment by dkoo** [comment]
Hi all, I've been involved with the `sync-newspack-blocks.sh` process in the past (but not for a while). From what I understand, WP.com uses the Homepage Posts and Post Carousel blocks that we maintain in the separate [Newspack Blocks](https://github.com/Automattic/newspack-blocks/) repository, and this script is part of what's used to sync code from that repository down to WP.com. That's a deliberate dev process and shouldn't cause any errors at runtime or if you're not trying to sync the code from our repository.

I did want to ask a higher-level question, though—the syncing was put in place because our blocks fulfilled a need in WP.com before the core Query block existed. Does it still make sense to include the Newspack blocks as part of WP.com, and if not, what would the process look like to deprecate that sync? On the other hand, if it does make sense to continue including them in WP.com, how can we improve the syncing process to make it less disruptive/more automated to your workflow?

**Review by adekbadek — COMMENTED** [review]
Our team does not use this process, if the change suits you, it's fine :)

File: apps/editing-toolkit/bin/sync-newspack-blocks.sh
Comment (jeyip) [reply apps/editing-toolkit/bin/sync-newspack-blocks.sh:137]: > From what I understand, WP.com uses the Homepage Posts and Post Carousel blocks that we maintain in the separate [Newspack Blocks](https://github.com/Automattic/newspack-blocks/) repository, and this script is part of what's used to sync code from that repository down to WP.com. That's a deliberate dev process

According to @dkoo, it sounds like this is a dev script.

To be extra safe though, I added an extra condition [676c50f](https://github.com/Automattic/wp-calypso/pull/65043/commits/676c50fdf2727402e37dbd0359dbe294265af873). I verified that we are in `MODE='npm'`  when running `yarn dev --sync` and _only_ skip the exiting logic in that case.  https://github.com/Automattic/wp-calypso/blob/89b10bccbb313c441e60d5ce0a7ec59dbba741d7/apps/editing-toolkit/package.json#L26

How does that sound to y'all for the time being to unblock folks, especially while, as Derek mentioned, we have broader discussions of whether or not we still want newspack blocks at all?

**Comment by jeyip** [comment]
> I did want to ask a higher-level question, though—the syncing was put in place because our blocks fulfilled a need in WP.com before the core Query block existed. Does it still make sense to include the Newspack blocks as part of WP.com, and if not, what would the process look like to deprecate that sync?

I think this is a fair point. I haven't spent much time with the query loop block, but from what I can tell there _does_ appear to be significant overlap with the newspack blog-posts + post carousel blocks.

As far as deprecation goes, I'd defer to @Automattic/team-calypso. To broach the discussion about the existence of Newspack blocks, I can leave a quick post in the dotcom p2.

Those resulting conversations sound like they can be follow-ups and we can move forward with this PR. Lemme know if that sounds okay @dkoo @creativecoder @mattwiebe

**Comment by jeyip** [comment]
Planning to merge this tomorrow and proceed with[ the discussed follow-up p2](https://github.com/Automattic/wp-calypso/pull/65043#issuecomment-1171600127) @creativecoder @fullofcaffeine

**Comment by jeyip** [comment]
I'll write up a P2 about this script deprecation tomorrow

**Comment by jeyip** [comment]
P2 post about deprecating Newspack Blocks and its syncing script as well paYE8P-1Jw-p2

File: apps/editing-toolkit/bin/sync-newspack-blocks.sh
Comment (creativecoder) [reply apps/editing-toolkit/bin/sync-newspack-blocks.sh:137]: Sorry for the late reply. Looks like I was mistaken and this isn't used by Newspack at all, so we're free to edit this as needed. In that case removing the exit line altogether seems fine, but I'm also good with the conditional check that you ended up with.

Thanks for landing this!

Comment (creativecoder) [reply apps/editing-toolkit/bin/sync-newspack-blocks.sh:137]: Update... it looks like throwing an error for ill-formatted code in the production build is still desired: paYE8P-1I4-p2#comment-1588

**Comment by noahtallen** [comment]
> Because our understanding is that sync-newspack-blocks.sh is a script meant for development purposes, in this PR, we remove logic that exits the syncing execution thread when a phpcbf error is thrown.

Also adding some context here from my p2 comment: the purpose of the script is to make sure that the source code from newspack-blocks in npm is copied to ETK's source code. The main production-oriented thing that phpcbf gives us is that it auto-fixes the textdomain to "full-site-editing," since the original text domain is related to newspack. 

As to _why_ this works the way it does... well, another team implemented it 3 years ago and it hasn't received that much love in the meantime. It was also important to use our webpack process and to kind of "copy out" the newspack files as needed rather than bundling the entire thing as-is. At the time, we did not want to use the Newspack brand for the blocks, as folks thought that would confuse customers. So they are re-registered in ETK to "feel" like normal WordPress.com blocks. This means our ETK JS needs to import the source code directly and also rewrite all the text domains.

Since you don't normally build blocks to be used in this really weird way, it's pretty hacky at each layer. But it has been working ok for the past few years!

Since PHP is also used by the blocks, it's important to copy that code over as well. And since we're copying the newspack source code, we need to make sure parts of it get updated to work in wpcom. This script makes sure all that stuff happens at build time. 

In terms of how to solve this problem: I believe we originally were originally running PHPCBF with a ruleset which _only_ included the textdomain fix. IIRC, WordPress.com's precommit hooks started detecting PHPCS errors in newspack, so we started running the autofixer with a broader ruleset. (I'm also pretty sure [I went into Newspack blocks](https://github.com/Automattic/newspack-blocks/pull/690), added phpcs to their CI, and fixed every error I could find... so I wonder why some are slipping through now!)

It's possible things have improved on wpcom so that it won't fail the precommit hook in the same way. If that's the case, we could move back to only fixing the textdomain

File: apps/editing-toolkit/bin/sync-newspack-blocks.sh
Comment (noahtallen) [inline apps/editing-toolkit/bin/sync-newspack-blocks.sh:138]: Since the build process does use the `npm` mode, I think this could allow an important issue to slip through in the production build.

That said, I'd still expect phpcbf to be formatting the textdomain correctly most of the time, so it probably won't practically have a huge impact unless the problem is that phpcbf just falls over entirely

Comment (jeyip) [reply apps/editing-toolkit/bin/sync-newspack-blocks.sh:138]: Would it be worthwhile to introduce some sort of flag identifying production builds ( or reuse some constant if it already exists )? Marcelo mentioned something about node env variables earlier.

Happy to do that if it guarantees we won't have important issues in the future

Comment (noahtallen) [reply apps/editing-toolkit/bin/sync-newspack-blocks.sh:138]: Good question! What would that help us solve here?

Comment (noahtallen) [reply apps/editing-toolkit/bin/sync-newspack-blocks.sh:138]: `NODE_ENV=production` might be set in the environment, but you might have to double check with `yarn build`. I'm not sure if that environment variable is passed down into the context when this script executes

Comment (jeyip) [reply apps/editing-toolkit/bin/sync-newspack-blocks.sh:138]: > What would that help us solve here?

I imagined we'd add some sort of condition to exit the build process if we're creating a production build _and_ the mode is `npm`

**Pull request merged**
